linksharing/handler.go

```diff
@@ -136,14 +152,52 @@ func (handler *Handler) serveHTTP(w http.ResponseWriter, r *http.Request) (err e
 	_, download := r.URL.Query()["download"]
 	_, view := r.URL.Query()["view"]
 	if !download && !view {
+		ipBytes, err := object.GetObjectIPs(ctx, uplink.Config{}, access, bucket, key)
+		if err != nil {
+			handler.handleUplinkErr(w, "get object IPs", err)
+			return err
+		}
+
+		var ipStrings []string
+		for index, _ := range ipBytes {
+			ip := string(ipBytes[index])
+			ipStrings = append(ipStrings, ip)
+		}
+
+		var ipInfos []objectmap.IPInfo
+		for _, address := range ipStrings {
+			info, err := handler.mapper.GetIPInfos(address)
+			if err != nil {
+				handler.log.Error("failed to get IP info", zap.Error(err))
+				continue
+			}
+
+			ipInfos = append(ipInfos, *info)
+		}
+
+		var locations []Location
+		for _, info := range ipInfos {
```

Comment:
no need to have 3 loops here, try something like this
```
		var locations []Location
		for _, ip := range ipBytes {
			info, err := handler.mapper.GetIPInfos(string(ip))
			if err != nil {
				handler.log.Error("failed to get IP info", zap.Error(err))
				continue
			}

			location := Location{
				Latitude:  info.Location.Latitude,
				Longitude: info.Location.Longitude,
			}

			locations = append(locations, location)
		}
```

Reply:
done